### PR TITLE
Make a number of small content improvements

### PR DESCRIPTION
**Pull request by StevenMaude**
This PR provides slight improvements to the documentation. There are:

* Small editorial changes to improve readability and clarity
* Relevant links added and existing broken links updated
* Slight reordering of documentation sections
* Several typo fixes

We may want to later review whether the existing format works well for
users, but this polishes what we have for now :sparkles:

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -40,17 +40,21 @@ See [git workflow](git-workflow.md) for advice about how to use git effectively.
...
   We use Docker containers to simulate the production environment on your machine, so you can [test code locally](actions-pipelines.md).
 
+## How should I start learning to use OpenSAFELY?
+
+The [Analysis workflow](workflow) gives a high-level view of the steps involved in an OpenSAFELY research project.
+
 See our [installation](install-intro.md) pages for complete installation instructions.
 
-We recommend following one of our OpenSAFELY walkthroughs (see [here](https://github.com/opensafely/os-demo-research#opensafely-demo-materials)) to guide you through the platform workflow, rather than using the documentation pages alone.
+We recommend following one of our OpenSAFELY walkthroughs (see the [notebook here](https://github.com/opensafely/os-demo-research#opensafely-demo-materials)) to guide you through the platform workflow on your own computer with dummy data, rather than using the documentation pages alone.
```
Comment (madwort) [inline docs/index.md:57]: I'm not a fan of `here` in link text - we could probably go further - how about `see the [opensafely demo notebook]`?

Comment (StevenMaude) [reply docs/index.md:57]: Agree :+1:

Comment (StevenMaude) [reply docs/index.md:57]: There are a few of these to fix throughout.

**Review by wjchulme — APPROVED** [review]
A much needed sweep, thanks.

File: docs/codelist-creation.md [hunk 2]
```diff
@@ -2,43 +2,43 @@
 
 The general workflow for creating codelists is as follows:
 
-1. Search [codelists.opensafely.org](https://codelists.opensafely.org) for codelists that meet or nearly meet your requirements and make sure that one doesn't already exist
-1. If you need to build a new codelist [sign up for an account on codelists.opensafely](https://codelists.opensafely.org/accounts/register/)
-1. Create a new issue on the [codelist-development repo](https://github.com/opensafely/codelist-development)
+1. Search [OpenCodelists](https://codelists.opensafely.org) for codelists that meet or nearly meet your requirements and make sure that one doesn't already exist.
```
Comment (madwort) [inline docs/codelist-creation.md:5]: we can put the new url here now?

Comment (StevenMaude) [reply docs/codelist-creation.md:5]: Yes, absolutely :slightly_smiling_face:

**Comment by StevenMaude** [comment]
Thanks @wjchulme for reviewing this :eyes:

As suggested, I'll update the OpenCodelists URLs, probably get round to this today, and then I'll merge, unless anyone else wants to highlight any further changes here.

**Comment by StevenMaude** [comment]
Updated the OpenCodelists URL, so merging :rocket:

**Pull request merged**